clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -52,13 +53,54 @@
 
     private final ConfigDef definition;
 
+    private static final String CONFIG_PROVIDERS_CONFIG = "config.providers";
+
+    /**
+     * Construct a configuration with a ConfigDef, the configuration properties, optional {ConfigProviders}.
+     *
+     * The originals is a name-value pair of config provider configs as well as configuration properties. The
+     * value of the configuration can be a variable as defined below or the actual value. This constructor will
+     * first instantiate the ConfigProviders using the config provider configs, then it will find all the
+     * variables in the values of the originals configurations, attempt to resolve the variables using the named
+     * ConfigProviders, and then parse and validate the configurations.
+     *
+     * ConfigProviders can be passed either as configs in the originals map or in the separate configProviderProps
+     * map. If config providers are passed in the configProviderProps any providers in originals map will be
+     * ignored. If ConfigProvider is not provided, the constructor will skip the variable substitution step and
```

Comment:
"If ConfigProvider is not provided" --> Strictly speaking, the parameters don't pass `ConfigProvider` instances, so this sentence is not very accurate. We need to be very careful about the terms we use, and we should not use the term "ConfigProvider" when we really mean "ConfigProvilder properties".